boring/src/x509/store.rs

```diff
@@ -78,6 +80,24 @@ impl X509StoreBuilderRef {
         unsafe { cvt(ffi::X509_STORE_add_cert(self.as_ptr(), cert.as_ptr())).map(|_| ()) }
     }
 
+    /// Adds a CRL to the certificate store.
+    ///
+    /// This corresponds to [`X509_STORE_add_crl`].
+    ///
+    /// [`X509_STORE_add_crl`]: https://www.openssl.org/docs/man1.1.1/man3/X509_STORE_add_crl
+    pub fn add_crl(&mut self, crl: X509CRL) -> Result<(), ErrorStack> {
```

Comment:
One annoying wart here:
You can iterate over the `X509Object`s in a store (see `X509StoreRef::objects`), which are either `X509 | X509CRL`. Boring exports the method to extract an `X509` from an `X509Object` (`X509_OBJECT_get0_X509`), but not `X509_OBJECT_get0_X509_CRL`.

As a result, you can only get the inner object out if it's an X509 when using an `X509Object`.

Reply:
It seems like we could add `X509Object::x509_crl` to live alongside `X509Object::x509`, no?

Reply:
Right, the problem is that I don't think it's possible to implement the rust method `X509Object ::x509_crl` against boringssl. OpenSSL 1.1.0 has `X509_OBJECT_get0_X509_CRL`, but it's not in boringssl. And the object struct (`x509_object_st`) is defined an internal header so we only have an opaque pointer.

Reply:
Ah, I see what you mean. I guess `X509_STORE_get1_certs` and `X509_STORE_get1_crls` will have to do.